### PR TITLE
Adds support for Stella_en_v5 embedding model - 1.5B variant

### LINKED ISSUE #2525: [QUESTION] Protocol of adding a new model (Stella_en_<*>_v5  family) implementation with Candle
Hi,

I have a working implementation of [Stella_en_<*>_v5](https://huggingface.co/dunzhang/stella_en_1.5B_v5) family of models which is one of the top ranking model in the MTEB leaderboard for reranking and retrieval.

It's basically built on top of the `candle-transformers::qwen2` implementation with the language modeling head swapped for their pre-trained dense layer.

I was hoping to open a pull request with candle along with an example.

**Questions**:
1. Would it be OK?
2. If it is, is there a protocol to follow for `candle_transformers`? Something akin to [How to add a model to `transformers`](https://huggingface.co/docs/transformers/en/add_new_model)
3. For `embedding` models are there any `candle` standard API implementations required? My implementations just spits out the logits from the forward pass.

Looking for some guidance. Thanks in advance.

CC:
@EricLBuehler @LaurentMazare

### PR DESCRIPTION
**Pull request by AnubhabB**
`Stella_en_1.5B_v5` is a top ranking model (top 5) in Retrieval and Reranking tasks as of 8th October 2024. 

[Model Card](https://huggingface.co/dunzhang/stella_en_1.5B_v5)

This PR adds support for the model along with some examples.

### Tasks:
- [x] Check license: Model is licensed `MIT`
- [x] Check numerical accuracy - Verified against source pytorch implementation. At `f32` candle implementation yields mathematically equivalent up to a 3-decimal places rounded .. beyond which we observe some variations. Code used to verify `numeric equivalence` can be found [here](https://github.com/AnubhabB/tensor-bench/tree/stella-num/src/stella_num).
- [x] Authors example from the model card added and reproduced.

#### Note:
`Stella_400M` seems to follow a significantly different implementation. Have not implemented it in this PR.

Closes #2525

### CODE REVIEW

**Review by LaurentMazare — APPROVED** [review]

**Comment by LaurentMazare** [comment]
Thanks!
